### PR TITLE
Make `UpdateSettingsClusterStateUpdateRequest` a record

### PR DESCRIPTION
**Pull request by DaveCTurner**
No need to extend `IndicesClusterStateUpdateRequest`, this thing can be
completely immutable.

### CODE REVIEW

**Review by DaveCTurner — COMMENTED** [review]
This one's a little more involved than the other similar changes I just opened because I've introduced some enums to avoid having two bare `boolean` parameters, and because there are some slightly more complex tests involving this class. Still hopefully not too hard to review.

**Review by nielsbauman — APPROVED** [review]

File: server/src/main/java/org/elasticsearch/upgrades/SystemIndexMigrator.java [hunk 1]
```diff
@@ -69,6 +69,9 @@
 public class SystemIndexMigrator extends AllocatedPersistentTask {
     private static final Logger logger = LogManager.getLogger(SystemIndexMigrator.class);
 
+    // TODO should this be longer and/or configurable?
+    private static final TimeValue SYSTEM_INDEX_MIGRATOR_MASTER_NODE_TIMEOUT = TimeValue.timeValueSeconds(30);
```
[the remark this answers is not in the document]
Comment (DaveCTurner) [reply server/src/main/java/org/elasticsearch/upgrades/SystemIndexMigrator.java:73]: No, that's for some future developer to find. It's been like this forever, there's no pressing need to fix it, this change is just turning the implicit lack of a timeout into an explicit check.

[the remark this answers is not in the document]
Comment (DaveCTurner) [reply server/src/main/java/org/elasticsearch/upgrades/SystemIndexMigrator.java]: Oh wait the default is `null` which means an infinite timeout here, not 30s like in `org.elasticsearch.action.support.master.MasterNodeRequest`. Oops. Fixed in d78bb19179367ccf895f5c286dd63e44ba4a688e.

**Comment by DaveCTurner** [comment]
Reverted due to conflicts in downstream projects, will try again in #113432.

**Pull request merged**